lib/packwerk/const_node_inspector.rb

```diff
@@ -40,5 +35,19 @@ def constant_in_module_or_class_definition?(node, parent:)
         parent_name && parent_name == Node.constant_name(node)
       end
     end
+
+    def fully_qualify_constant(node, ancestors:)
```

Comment:
Extracting these private methods is a good idea. I would love to suggest better names for them to more clearly describe what they’re doing, but I actually don’t understand what the code does in the first place (why is it sometimes returning a “parent module name” generated only from the ancestor nodes, but sometimes a ”qualified constant” name generated from the node itself?) so can’t begin to think of anything better. Fixing the deeper problem is a job for another PR. 😄

Reply:
Agreed, I'm not confident in why the code does what it does. I think it would be good to go through the test cases and document how the data gets manipulated in those methods. I would add to this that we can probably extract those private methods into an Ancestor module in a separate PR and document it then. WDYT?

Reply:
Again, it’s a good idea — let’s decide separately whether it’s worth doing in another PR.